torch/csrc/jit/codegen/cuda/lower_loops.cpp

```diff
@@ -159,7 +160,12 @@ void LoopNestGenerator::handle(Expr* expr) {
       openFor(loop);
     }
   }
-
+  if (lower_mma_utils::isMMAMultiply(expr)) {
+    // This is the complication from mapping 2 fusionIR ops into 1
+    //  kernel IR op, i.e. multiply+reduce -> mma
+    // The mma op will be generated when lowering the reduction op.
+    return;
+  }
```

Comment:
This isn't related to loop generation at all. Could this be done elsewhere?

Reply:
I think the only thing the above line is doing is to avoid lowering the multiply op from fusion IR. Which happens just the line below. So the only reason this is here is because we generally lower the expressions here in the loop nest generation. We could have more discussion on generally separating `loopnest` and `expr lower` though if we need.

Reply:
Yeah, I understand that, but as a software engineering point of view, I'd like to have this clearly separated from other logic. Perhaps, having a separate function like, `shouldSkipExpression(Expr*)`, and make `LoopNestGenerator::handle(Expr*)` look like:

```
void LoopNestGenerator::handle(Expr* expr) {
   if (shouldSkipExpression(expr)) {
        return;
    }
     // the same function body 
     ...
}
```

Another option may be to add a new KIR expression, `Nop`, to represent no-op expressions and map the MMA multiply expressions to `kir::Nop`.